agents/agents-ext/src/commonMain/kotlin/ai/koog/agents/ext/tool/file/ReadFileTool.kt

```diff
@@ -53,11 +53,15 @@ public class ReadFileTool<Path>(private val fs: FileSystemProvider.ReadOnly<Path
      * @property file the file entry containing metadata and content
      */
     @Serializable
-    public data class Result(val file: FileSystemEntry.File) : ToolResult.TextSerializable() {
+    public data class Result(
+        val file: FileSystemEntry.File,
+        internal val warningMessage: String? = null
```

Comment:
Generally it looks a bit weird when public data class has non-public fields. I see no reason to make it internal here, let's remove `internal`?